daemon/stats.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef STATS_H
+#define STATS_H
+
+#include <string>
+
+namespace monitord {
+namespace server {
+
+class cpu_monitor
+{
+public:
+  cpu_monitor(const cpu_monitor&) = delete;
```

Comment:
Nitpick: The less error prone approach would be in defining the class `final`, as you haven't defined the virtual dtor.
The same can be applied for all classes in the project.